src/esp/physics/PhysicsObjectBase.h

```diff
@@ -401,6 +411,28 @@ class PhysicsObjectBase : public Magnum::SceneGraph::AbstractFeature3D {
    */
   const assets::ResourceManager& resMgr_;
 
+  /** @brief Accessed internally. Get an appropriately cast copy of the @ref
+   * metadata::attributes::SceneObjectInstanceAttributes used to place the
+   * object within the scene.
+   * @return A copy of the initialization template used to create this object
+   * instance or nullptr if no template exists.
+   */
+  template <class T>
+  std::shared_ptr<T> getSceneInstanceAttrInternal() const {
```

Comment:
This is here to cater to Articulated Objects, which use a SceneAOInstanceAttributes, which inherits from SceneObjectInstanceAttributes. Nice to have access to the instance attributes cast appropriately for immediate access.